protocol/messenger.go

```diff
@@ -920,20 +917,7 @@ func (m *Messenger) Start() (*MessengerResponse, error) {
 		}
 	}
 
-	joinedCommunities, err := m.communitiesManager.Joined()
-	if err != nil {
-		return nil, err
-	}
-
-	for _, joinedCommunity := range joinedCommunities {
-		// resume importing message history archives in case
-		// imports have been interrupted previously
-		err := m.resumeHistoryArchivesImport(joinedCommunity.ID())
-		if err != nil {
-			return nil, err
-		}
-	}
-	m.enableHistoryArchivesImportAfterDelay()
+	m.startHistoryArchivesImportLoop()
```

Comment:
Keep the function simple, no need to nest the `go ...` inside
```go
func (m *Messenger) historyArchivesImportLoop() {
    defer gocommon.LogOnPanic()
    joinedCommunities, err := m.communitiesManager.Joined()
    // ...
}
```

And then start as just:

```suggestion
	go m.historyArchivesImportLoop()
```

Reply:
fixed